src/patterns/pages/workspace/data/workspace.service.ts

```diff
@@ -0,0 +1,214 @@
+import { Injectable } from '@angular/core';
+import { SampleOppData } from './datasource';
+import { Observable } from 'rxjs';
+
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class WorkspaceService {
+
+  constructor() { }
+
+  //Will pass in 
+  getData(filter: filter) {
+
+    let data = <Opportunity[]>SampleOppData;
+    let ob = Observable.of(data);
+    ob = this.filter(ob, filter);
+    let pageSize = 10;
+
+    return ob;
+  }
+
+
+  private filter(data: Observable<Opportunity[]>, filter: filter) {
+
+    if (filter.type) {
+      data = data.map(projects => projects.filter(proj => proj.data.type === filter.type));
+    }
+
+    if (filter.status) {
+      data = data.map(projects => projects.filter(proj => proj.status === filter.status));
+    }
+
+    if (filter.title) {
+      data = data.map(projects => projects.filter(proj => proj.data.title.indexOf(filter.title) !== -1));
+    }
+
+    return data;
+  }
+
+  sort(data: Opportunity[], filter: filter) {
+
+
+  }
+}
+
+export class filter {
+
+
+  //zero based
+  page: number;
+  type: string;
+  status: string;
+  title: string;
+
+
+
+
+
+}
+
+
+
+
+
+export interface City {
+  code?: string;
+}
+
+export interface State {
+  code?: string;
+}
+
+export interface Country {
+  code?: string;
+}
+
+export interface Location {
+  zip?: string;
+  city?: City;
+  state?: State;
+  country?: Country;
+  streetAddress?: string;
+}
+
+export interface Awardee {
+  name?: string;
+  duns?: string;
+  location?: Location;
+}
+
+export interface FairOpportunity {
```

Comment:
@thomastighe This still needed?

Reply:
Can you please explain which part you think is not needed.